tests/unit/test_logwriter.c

```diff
@@ -177,20 +190,27 @@ Test(logwriter, test_logwriter)
   configuration = cfg_new_snippet();
   LogWriterTestCase test_cases[] =
   {
-    {MSG_SYSLOG_STR, TRUE, NULL, LW_SYSLOG_PROTOCOL, EXPECTED_MSG_SYSLOG_STR},
-    {MSG_SYSLOG_STR, TRUE, "$MSGID $MSG", LW_SYSLOG_PROTOCOL, EXPECTED_MSG_SYSLOG_STR_T},
-    {MSG_SYSLOG_EMPTY_STR, TRUE, NULL, LW_SYSLOG_PROTOCOL, EXPECTED_MSG_SYSLOG_EMPTY_STR},
-    {MSG_SYSLOG_EMPTY_STR, TRUE, "$MSGID", LW_SYSLOG_PROTOCOL, EXPECTED_MSG_SYSLOG_EMPTY_STR_T},
-    {MSG_SYSLOG_STR, TRUE, NULL, LW_FORMAT_PROTO, EXPECTED_MSG_SYSLOG_TO_BSD_STR},
-    {MSG_SYSLOG_STR, TRUE, NULL, LW_FORMAT_FILE, EXPECTED_MSG_SYSLOG_TO_FILE_STR},
-    {MSG_BSD_STR, FALSE, NULL, LW_FORMAT_FILE, EXPECTED_MSG_BSD_STR},
-    {MSG_BSD_STR, FALSE, "$PRI $PID $MSG", LW_FORMAT_FILE, EXPECTED_MSG_BSD_STR_T},
-    {MSG_BSD_STR, FALSE, NULL, LW_FORMAT_PROTO, EXPECTED_MSG_BSD_TO_PROTO_STR},
-    {MSG_BSD_STR, FALSE, NULL, LW_SYSLOG_PROTOCOL, EXPECTED_MSG_BSD_TO_SYSLOG_STR},
-    {MSG_BSD_EMPTY_STR, FALSE, NULL, LW_FORMAT_PROTO, EXPECTED_MSG_BSD_EMPTY_STR},
-    {MSG_BSD_EMPTY_STR, FALSE, "$PID", LW_FORMAT_PROTO, EXPECTED_MSG_BSD_EMPTY_STR_T},
-    {MSG_ZERO_PRI, FALSE, NULL, LW_FORMAT_PROTO, EXPECTED_MSG_ZERO_PRI_STR},
-    {MSG_ZERO_PRI, FALSE, "$PRI", LW_FORMAT_PROTO, EXPECTED_MSG_ZERO_PRI_STR_T},
+    {MSG_SYSLOG_STR, TRUE, NULL, LW_SYSLOG_PROTOCOL, 0, EXPECTED_MSG_SYSLOG_STR},
+    {MSG_SYSLOG_STR, TRUE, "$MSGID $MSG", LW_SYSLOG_PROTOCOL, 0, EXPECTED_MSG_SYSLOG_STR_T},
+    {MSG_SYSLOG_STR, TRUE, "$MSGID $MSG", LW_SYSLOG_PROTOCOL, 61, EXPECTED_MSG_SYSLOG_STR_T_TRUNCATE},
+    // test that truncate doesn't apply on smaller messages
+    {MSG_SYSLOG_STR, TRUE, "$MSGID $MSG", LW_SYSLOG_PROTOCOL, 65507, EXPECTED_MSG_SYSLOG_STR_T},
+    {MSG_SYSLOG_EMPTY_STR, TRUE, NULL, LW_SYSLOG_PROTOCOL, 0, EXPECTED_MSG_SYSLOG_EMPTY_STR},
+    {MSG_SYSLOG_EMPTY_STR, TRUE, "$MSGID", LW_SYSLOG_PROTOCOL, 0, EXPECTED_MSG_SYSLOG_EMPTY_STR_T},
+    {MSG_SYSLOG_STR, TRUE, NULL, LW_FORMAT_PROTO, 0, EXPECTED_MSG_SYSLOG_TO_BSD_STR},
+    {MSG_SYSLOG_STR, TRUE, NULL, LW_FORMAT_FILE, 0, EXPECTED_MSG_SYSLOG_TO_FILE_STR},
+    {MSG_BSD_STR, FALSE, NULL, LW_FORMAT_FILE, 0, EXPECTED_MSG_BSD_STR},
+    {MSG_BSD_STR, FALSE, NULL, LW_FORMAT_FILE, 39, EXPECTED_MSG_BSD_STR_TRUNCATE},
+    {MSG_BSD_STR, FALSE, NULL, LW_FORMAT_FILE, 65507, EXPECTED_MSG_BSD_STR},
+    {MSG_BSD_STR, FALSE, "$PRI $PID $MSG", LW_FORMAT_FILE, 0, EXPECTED_MSG_BSD_STR_T},
+    {MSG_BSD_STR, FALSE, "$PRI $PID $MSG", LW_FORMAT_FILE, 23, EXPECTED_MSG_BSD_STR_T_TRUNCATE},
```

Comment:
Optional:
I think the value of `truncate_size` should be close to the declaration of `EXPECTED_MSG_*`.
It might also be possible to calculate `truncate_size` based on the expected output (`strlen`, for example), or vice versa.

Reply:
Okay, I agree. It's flexible to have e.g. `strlen`.